text/1003-cli-naming.rst

```diff
@@ -19,110 +19,238 @@ specified.
 
-The last observation is what prompted the creation of this RFC.  We either
-need to update RFC 1001 to use the ``<action>-<category>`` scheme or
-adopt the ``<group> <command>`` for all other commands.  Combining two
-drastically different CLI designs in one tool is not an option.
+**Concern**: The ``<action>-<category>`` scheme does not allow
+category-specific help sections. With ``<group> <command>`` scheme, top-level
+help sections are natural: ``edgedb migration -h`` command can have a
+description of the recommended migrations workflow, while
+``edgedb server -h`` would explain to the user how to work with EdgeDB
+servers.
 
+**Decision**: the CLI is an unlikely place for a user to learn about new
+complex functionality like the migration subsystem. Better suited avenues
+for this are the online documentation and/or ``man`` pages, or better yet,
+``tldr``.
 
-Overview
-========
+To address discoverability of related sub-commands, top-level listing of
+sub-commands should not be alphabetical but rather grouped by functionality.
 
-The RFC proposes to adopt the ``<group> <command>`` naming scheme for
-the ``edgedb`` CLI.  Here's an outline of current and future ``edgedb``
-subcommands:
+``configure`` uses ``<group> <command>`` whereas ``list-`` is flat
+------------------------------------------------------------------
+
+**Concern**: ``configure`` is inconsistent with ``list-``.
+
+**Decision**: It should be replaced by ``edgedb insert-config``,
+``edgedb set-config``, and``edgedb reset-config``.
+
+Why is ``server`` special?
+--------------------------
+
+**Concern**: the ``edgedb server`` CLI design outlined in RFC 1001
+directly conflicts with the ``<action>-<category>`` scheme.
+
+**Decision**: this sub-system is different because it is in fact
+a tool within a tool. Whereas other commands deal with a single
+database server, often a single database within that server, the
+``edgedb server`` sub-system deals with management of potentially
+many databases and server instances. With that in mind, keeping it
+separate makes sense.
+
+In particular, trying to shoehorn the sub-commands of ``edgedb server``
+into the ``<action>-<object>`` scheme would essentially move all of
+them into flat ``*-instance`` (or ``*-server``) sub-commands.
+
+Are ``list-*`` commands useful?
+-------------------------------
+
+**Concern**: there are nine ``list-*`` sub-commands which is over 36%
+of the available sub-commands. They mirror REPL functionality for listing
+items in the current database, for example ``\la`` or ``\list-aliases``
+will list available aliases. Are all of them necessary?
+
+**Decision**: the CLI should keep listings of entities that it can
+directly manipulate with other sub-commands. This includes:
+
+* ``list-databases`` (through ``create-database``);
+
+* ``list-ports`` (through ``configure``); and
+
+* ``list-roles`` (through ``alter-role``).
+
+Other listings should be removed as they are redundant with REPL
```

Comment:
ACTIONABLE ITEM 5: remove most list-* commands.

Reply:
+1. At least until there's a clear user request/reason to have them exposed in CLI and not just in REPL.

Reply:
We can hide them from the help for brewity, if you wish. But I don't see the reason to remove them. `edgedb list-object-types` can be used in CI to find out what's going on. Also `edgedb list-object-types | cat` emits tab-separated data that can be used for scripting.

Reply:
> edgedb list-object-types can be used in CI to find out what's going on.

Or you could just pipe an introspection query to the repl. But I hear you.

Getting back to the topic of `-h` and `--help` being slightly different -- maybe we omit them from `-h` but show everything in `--help`?

Also, @elprans, please join.

Reply:
There's still another suggestion @1st1 made during the meeting: keep them but make them different from \list-* in REPL: single-tab separated to be easily machine parseable with Unix pipelines.

Reply:
(Keeping this as unresolved for now.)

Reply:
Observation: not listing sub-commands alphabetically, per another accepted decision in the RFC, would help isolating the `list-*` sub-commands and so it might be enough in terms of `--help` readability. And we want to keep them to make machine-parseable versions of what you're getting in the REPL.